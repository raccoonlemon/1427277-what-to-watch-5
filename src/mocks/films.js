export const films = [
  {
    id: `1`,
    title: `The Grand Budapest Hotel`,
    genre: `drama`,
    year: 2014,
    director: `Wes Andreson`,
    duration: 124,
    starring:
    [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
    Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    posterSmall: `img/the-grand-budapest-hotel-poster.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `2`,
    title: `Pulp fiction`,
    genre: `crime`,
    year: 1994,
    director: `Quentin Tarantino`,
    duration: 118,
    starring:
    [
      `Tim Roth`,
      `Amanda Plummer`,
      `Laura Lovelace`,
      `John Travolta`,
      `Samuel L. Jackson`,
      `Phil LaMarr`,
      `Frank Whaley`,
      `Burr Steers`,
      `Bruce Willis`
    ],
    description: `The lives of two mob hitmen, a boxer, a gangster and his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    background: `img/pulp-fiction.jpg`,
    poster: `img/pulp-fiction.jpg`,
    posterSmall: `img/pulp-fiction.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `3`,
    title: `Moonrise Kingdom`,
    genre: `romance`,
    year: 2012,
    director: `Wes Anderson`,
    duration: 98,
    starring:
    [
      `Bruce Willis`,
      `Edward Norton`,
      `Bill Murray`,
      `Frances McDormand`,
      `Tilda Swinton`,
      `Jared Gilman`,
      `Kara Hayward`,
      `Jason Schwartzman`,
      `Bob Balaban`
    ],
    description: `A pair of young lovers flee their New England town, which causes a local search party to fan out to find them.`,
    background: `img/moonrise-kingdom.jpg`,
    poster: `img/moonrise-kingdom.jpg`,
    posterSmall: `img/moonrise-kingdom.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `4`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `fantasy`,
    year: 2018,
    director: `David Yates`,
    duration: 134,
    starring:
    [
      `Johnny Depp`,
      `Kevin Guthrie`,
      `Carmen Ejogo`,
      `Wolf Roth`,
      `Eddie Redmayne`,
      `Zoë Kravitz`,
      `Callum Turner`,
      `Derek Riddell`,
      `Cornell John`
    ],
    description: `The second installment of the "Fantastic Beasts" series featuring the adventures of Magizoologist Newt Scamander.`,
    background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    posterSmall: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `5`,
    title: `Aviator`,
    genre: `biography`,
    year: 2004,
    director: `Martin Scorsese`,
    duration: 110,
    starring:
    [
      `Leonardo DiCaprio`,
      `Cate Blanchett`,
      `Kate Beckinsale`,
      `John C. Reilly`,
      `Alec Baldwin`,
      `Alan Alda`,
      `Jude Law`,
    ],
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    background: `img/aviator.jpg`,
    poster: `img/aviator.jpg`,
    posterSmall: `img/aviator.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: `6`,
    title: `Shutter Island`,
    genre: `thriller `,
    year: 2010,
    director: `Martin Scorsese`,
    duration: 142,
    starring:
    [
      `Leonardo DiCaprio`,
      `Mark Ruffalo`,
      `Ben Kingsley`,
      `Michelle Williams`,
      `Emily Mortimer`,
      `Patricia Clarkson`,
      `Max von Sydow`,
    ],
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer who escaped from a hospital for the criminally insane.`,
    background: `img/shutter-island.jpg`,
    poster: `img/shutter-island.jpg`,
    posterSmall: `img/shutter-island.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `7`,
    title: `Johnny English`,
    genre: `comedy `,
    year: 2003,
    director: `Peter Howitt`,
    duration: 109,
    starring:
    [
      `Rowan Atkinson`,
      `Natalie Imbruglia`,
      `Ben Miller`,
      `John Malkovich`,
      `Eddie Redmayne`,
      `Zoë Kravitz`,
      `Callum Turner`,
      `Derek Riddell`,
    ],
    description: `After a cyber-attack reveals the identity of all of the active undercover agents in Britain, Johnny English (Rowan Atkinson) is forced to come out of retirement to find the mastermind hacker.`,
    background: `img/johnny-english.jpg`,
    poster: `img/johnny-english.jpg`,
    posterSmall: `img/johnny-english.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },

  {
    id: `8`,
    title: `Seven Years in Tibet`,
    genre: `drama`,
    year: 1997,
    director: `Jean-Jacques Annaud`,
    duration: 125,
    starring:
    [
      `Brad Pitt`,
      `David Thewlis`,
      `BD Wong`,
      `Danny Denzongpa`,
      `Mako`,
      `Jamyang Jamtsho Wangchuk`,
      `Lhakpa Tsamchoe`,
      `Jetsun Pema`,
    ],
    description: `True story of Heinrich Harrer, an Austrian mountain climber who became friends with the Dalai Lama at the time of China's takeover of Tibet.`,
    background: `img/seven-years-in-tibet.jpg`,
    poster: `img/seven-years-in-tibet.jpg`,
    posterSmall: `img/seven-years-in-tibet.jpg`,
    video: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
];

export const promoFilm = films[0];
