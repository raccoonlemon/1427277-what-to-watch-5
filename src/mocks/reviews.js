const texts = [
  `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
  `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight.`,
  `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
  `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`
];

export const reviews = [
  {
    id: 1,
    date: new Date(2020, 10, 10, 0, 0, 0, 0),
    author: `Ellie`,
    rating: 10,
    text: texts[0],
  },
  {
    id: 2,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Max`,
    rating: 5,
    text: texts[1]
  },
  {
    id: 3,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Parvati`,
    rating: 8,
    text: texts[2]
  },
  {
    id: 4,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Felix`,
    rating: 8,
    text: texts[3]
  },
  {
    id: 5,
    date: new Date(2020, 10, 12, 0, 0, 0, 0),
    author: `Nyoka`,
    rating: 4,
    text: texts[4]
  }
];
