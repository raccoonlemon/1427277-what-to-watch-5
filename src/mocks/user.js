export const user = {
  id: `1`,
  email: `email@email.com`,
  name: `User Name`,
  avatarSrc: `src`,
};
