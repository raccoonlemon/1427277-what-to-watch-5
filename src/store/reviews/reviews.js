import {RequestStatus} from "../../const";
import {extend} from "../../utils/common";
import {NameSpace} from "../namespace";

export const ActionType = {
  LOAD_REVIEWS: `LOAD_REWIEVS`,
  REVIEW_POST_REQUESTED: `REVIEW_POST_REQUESTED`,
  REVIEW_POST_RECIEVED: `REVIEW_POST_RECIEVED`,
  REVIEW_POST_FAILED: `REVIEW_POST_FAILED`,
  SET_POSTED_REVIEW: `SET_POSTED_REVIEW`,
};

const initialState = {
  requestStatus: RequestStatus.NOT_REQUESTED,
  errorCode: 0,
  postedReviews: [],
  reviews: [],
};

// Reducer

export const reviewsReducer = (state = initialState, action) => {
  switch (action.type) {

    case ActionType.LOAD_REVIEWS:
      return extend(state, {reviews: action.payload});

    case ActionType.REVIEW_POST_REQUESTED:
      return extend(state, {requestStatus: RequestStatus.REQUESTED});

    case ActionType.REVIEW_POST_FAILED:
      return extend(state, {requestStatus: RequestStatus.REQUEST_FAILED, errorCode: action.payload});

    case ActionType.REVIEW_POST_RECIEVED:
      return extend(state, {requestStatus: RequestStatus.RECIEVED});

    case ActionType.SET_POSTED_REVIEW:
      return extend(state, {postedReviews: action.payload});
  }

  return state;
};

// Actions

export const loadReviews = (reviews) => ({
  type: ActionType.LOAD_REVIEWS,
  payload: reviews,
});

export const reviewPostRequested = () => ({
  type: ActionType.REVIEW_POST_REQUESTED,
  payload: {},
});

export const reviewPostRecieved = () => ({
  type: ActionType.REVIEW_POST_RECIEVED,
  payload: {},
});

export const reviewPostFailed = (error) => ({
  type: ActionType.REVIEW_POST_FAILED,
  payload: error,
});

export const setPostedReview = (review) => ({
  type: ActionType.SET_POSTED_REVIEW,
  payload: review,
});

// Selectors

const nameSpace = NameSpace.REVIEWS;

export const selectReviewRequestStatus = (state) => state[nameSpace].requestStatus;
export const selectIsReviewPostRequested = (state) => selectReviewRequestStatus(state) === RequestStatus.REQUESTED;
export const selectIsReviewPostFailed = (state) => selectReviewRequestStatus(state) === RequestStatus.REQUEST_FAILED;
export const selectReviewErrorCode = (state) => state[nameSpace].errorCode;
export const selectReviews = (state) => state[nameSpace].reviews;
