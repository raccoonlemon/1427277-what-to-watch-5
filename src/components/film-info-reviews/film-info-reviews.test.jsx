import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {reviews} from "../../mocks/reviews";
import FilmInfoReviews from "./film-info-reviews";

const renderer = new ShallowRenderer();

it(`<FilmInfoReviews> renders correctly`, () => {
  renderer.render(<FilmInfoReviews reviews={reviews}/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
