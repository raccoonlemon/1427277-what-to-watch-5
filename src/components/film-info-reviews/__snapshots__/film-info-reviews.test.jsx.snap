// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmInfoReviews> renders correctly 1`] = `
<React.Fragment>
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Ellie
            </cite>
            <time
              className="review__date"
              dateTime="2020-11-10T00:00"
            >
              November 10, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          10
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Max
            </cite>
            <time
              className="review__date"
              dateTime="2020-11-12T00:00"
            >
              November 12, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          5
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Parvati
            </cite>
            <time
              className="review__date"
              dateTime="2020-11-12T00:00"
            >
              November 12, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
    </div>
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Felix
            </cite>
            <time
              className="review__date"
              dateTime="2020-11-12T00:00"
            >
              November 12, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          8
        </div>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Nyoka
            </cite>
            <time
              className="review__date"
              dateTime="2020-11-12T00:00"
            >
              November 12, 2020
            </time>
          </footer>
        </blockquote>
        <div
          className="review__rating"
        >
          4
        </div>
      </div>
    </div>
  </div>
</React.Fragment>
`;
